### DISCUSSION TITLE
Should Homebrew (offer to) add an entry in /etc/paths.d on macOS?

### DISCUSSION THREAD

**Discussion by ratkins** [Homebrew/discussions#448] [Everyday usage]
The correct way to [add a path to all shells for all users on macOS](https://unix.stackexchange.com/questions/342574/use-etc-paths-or-etc-paths-d-to-add-items-to-the-path-in-macos-sierra) is to add an entry into `/etc/paths.d`.

Should the Homebrew installer suggest this or offer to do this as part of its installation process? That directory is not protected by SIP but is not world-writable, so this would require sudo.

It seems there is a lot of confusion about this by Homebrew users on M1 and the standard advice ("Add `path+=/opt/homebrew/bin` to `.zshrc`!") is not entirely correct (should be `.zshenv`) and incomplete, if you're not using zsh.

**Comment by SMillerDev** [comment]
The idea isn't to add the path for all users since homebrew only supports single user setups. 

If you have a suggestion for a better place to put the path definition, please open a pull request to the install script.

**Reply by ratkins** [reply]
Oh! That was a detail of which I was unaware.

In any case, I'll do the PR thing for `/etc/paths.d` if there's a likelihood it'll be accepted.

**Reply by jpedro** [reply]
In my case the directory `/etc/paths.d` gets loaded before `/etc/paths`:

```shell
$ cat /etc/paths
/usr/local/bin
/System/Cryptexes/App/usr/bin
/usr/bin
/bin
/usr/sbin
/sbin
```

which screws up a few things. For example, in this form `bash` get loaded from `/bin/bash` not `/opt/homebrew/bin/bash`.

```shell
$ sw_vers
ProductName:		macOS
ProductVersion:		26.1
BuildVersion:		25B78
```
